File: 413. Arithmetic_Slices/solution(improved).java
class Solution {
    public int numberOfArithmeticSlices(int[] nums) {
        
        if(nums.length < 3)
            return 0;
                
        int ans = 0;
        
        int hist_dif = nums[1] - nums[0];
        int count = 1;
        for(int i = 2 ; i < nums.length ; i++){
            if(nums[i]  - nums[i - 1] != hist_dif){
                hist_dif = nums[i] - nums[i - 1];
                count = 1;
            }
            else{
                count++;
                if(count >= 2)
                    ans += (count - 1);
            }
        }
        
        
        return ans;

    }
}

File: 2342. Max_Sum_of_a_Pair_With_Equal_Sum_of_Digits/solution(improved).cpp
class Solution {
public:
    int maximumSum(vector<int>& nums) {
        
        int dict[100];
        memset(dict, 0, sizeof(dict));
        
        int ans = -1;
        
        for(int i = 0 ; i < nums.size() ; i++){
            
            int cur_int = nums[i];
            int digit_sum = 0;
            while(cur_int > 0){
                digit_sum += cur_int % 10;
                cur_int /= 10;
            }
            
            if(dict[digit_sum] != 0){
                ans = max(ans, dict[digit_sum] + nums[i]);
                if(dict[digit_sum] < nums[i])
                    dict[digit_sum] = nums[i];
            }
            else
                dict[digit_sum] = nums[i];
            
        }
        
        
        return ans;
        
    }
};